src/EnergyPlus/api/datatransfer.cc

```diff
@@ -105,8 +104,13 @@ const char * listAllAPIDataCSV() {
         output.append(variable.VarNameOnly).append(",");
         output.append(variable.KeyNameOnlyUC).append("\n");
     }
-    // add output vars and meters
-    return output.c_str();
+    // note that we cannot just return a c_str to the local string, as the string will be destructed upon leaving
+    // this function, and undefined behavior will occur.
+    // instead make a small copy -- I realize it could cause its own issues, but I would not expect users to be making
+    // many many calls to this.  And if they did, just free the memory and be done.
+    char *p = new char[std::strlen(output.c_str())];
```

Comment:
@Myoldmopar This is going to be ripe for creating a memory leak. The caller needs to call `delete` on this allocated memory. Does Python take ownership of the returned `const char *`? I know in your unit tests there is a memory leak when implicitly converting to `std::string`.